src/main/java/it/gov/pagopa/payment/service/payment/barcode/BarCodeCreationServiceImpl.java

```diff
@@ -105,14 +104,13 @@ private void checkWallet(String initiativeId, String userId){
         WalletDTO wallet = walletConnector.getWallet(initiativeId, userId);
 
         if (wallet.getAmount().compareTo(BigDecimal.ZERO) == 0) {
-            throw new ClientExceptionWithBody(HttpStatus.FORBIDDEN,
+            throw new BudgetExhaustedException(
```

Comment:
some Exception, like this, could have a default code that is supposed to be always the same: we could thus expose a constructor without CODE, using such default code in the constructor